src/components/Api.js

```diff
@@ -0,0 +1,114 @@
+export default class API {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error ${res.status}`);
+    }
+  }
+
+  _request(url, options) {
+    return fetch(url, options).then(this._handleResponse);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).then(this._handleResponse);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    })
+      .then(this._handleResponse)
+      .then((data) => {
+        return data;
+      })
+      .catch((error) => {
+        console.error("Error in fetch:", error); // Debug line
+        throw error; // Optional: re-throw to propagate error
+      });
+  }
+
+  updateProfileInfo(data) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: data.title,
+        about: data.description,
+        avatar: data.avatar,
+      }),
+    }).then(this._handleResponse);
+  }
+
+  addCard({ name, link }) {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({
+        name,
+        link,
+      }),
+    }).then(this._handleResponse);
+  }
+
+  deleteCard(cardId) {
+    return fetch(`${this._baseUrl}/cards/${cardId}`, {
+      method: "DELETE",
+      headers: this._headers,
+    })
+      .then(this._handleResponse)
+      .catch((err) => {
+        console.error(err);
```

Comment:
* You haven't corrected my remark https://github.com/queenhekate/se_project_aroundtheus/pull/1/files#r1779494365